### PR TITLE
Remove support for Ruby v1.9.3

### LINKED ISSUE #902: Remove support for Ruby 1.9.3 and below
Any objections?

### PR DESCRIPTION
**Pull request by brasmusson**
## Summary

Remove support for Ruby v1.9.3. Closes #902.

## Details

Remove Ruby v1.9.3 from the Travis matrix. Remove the backport of Hash({...}).

### CODE REVIEW

**Comment by brasmusson** [comment]
We have to find a way to filer out the "`jruby: warning: unknown property jruby.cext.enabled`" warnings that the non v1.9.3 JRubies prints on Travis, a problem that also surfaces for [the Gherkin project](https://github.com/cucumber/gherkin/blob/master/ruby/Makefile#L41-L43).

**Comment by mattwynne** [comment]
@brasmusson I realised after I merged this that you'd also removed the other Rubies (2.0, 2.1, 2.2) from the build matrix. Was that deliberate? Are we assuming that testing 2.3 covers them all? Or should we add them back?

**Comment by brasmusson** [comment]
@mattwynne If you look at the list of commits, you will see "Try and fix JRuby error on Travis" twice and both times the build failed. When trying to fix JRuby I removed almost all other builds from the matrix to not waste computing power, but since I did not finally successfully fixed the JRuby build, I had not cleaned up this PR. What I really wanted say with my comment on #1009, was "yep, I have also run into the `unknown property jruby.cext.enabled`, and I worked on them here, but as you can see, I did not manage to get the JRuby build completely passed before I gave up."

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
-rvm:
-  - 2.3.0
-  - 2.2
-  - 2.1
-  - 2.0
-  - 1.9.3
-  - jruby-1.7.12
+matrix:
+  include:
+    - rvm: 2.3.0
```
Comment (mathieujobin) [inline .travis.yml]: this should be 2.3.1 or just 2.3

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,19 @@
 sudo: false
 
-rvm:
-  - 2.3.0
-  - 2.2
-  - 2.1
```
Comment (mathieujobin) [inline .travis.yml]: should keep 2.1 and 2.2 at least

File: .travis.yml (hunk 1 above)
Comment (brasmusson) [inline .travis.yml]: Previously just 2.3 did not work on Travis https://github.com/cucumber/cucumber-ruby/commit/d688d886a60dc51973c2fb572a277be4b9720b3a.

Comment (mattwynne) [inline .travis.yml]: Ugh yup: https://travis-ci.org/cucumber/cucumber-ruby/jobs/153563189

**Pull request merged**